src/Context/ColorScheme.tsx

```diff
@@ -0,0 +1,73 @@
+import {
+    createContext,
+    useContext,
+    useState,
+    ReactNode,
+    useEffect,
+} from "react";
+
+type ColorSchemeContextType = {
+    colorScheme: "dark" | "light";
+    toggleColorScheme: () => void;
+};
+
+const ColorSchemeContext = createContext<ColorSchemeContextType | undefined>(
+    undefined
+);
+
+export function ColorSchemeProvider({ children }: { children: ReactNode }) {
```

Comment:
There is another project where I use something like this for managing the color scheme in a context. 

```
 const prefersDarkMode = useMediaQuery('(prefers-color-scheme: dark)'); // default to user's current color scheme ==> window.matchMedia("(prefers-color-scheme: dark)").matches

  const [mode, setMode] = useState<PaletteMode>(prefersDarkMode ? 'dark' : 'light');

  const handleTheme = () => {
    setMode((prevMode) => (prevMode === 'light' ? 'dark' : 'light'));
  };

```

This you shouldn't need a useEffect.

Reply:
Thanks a lot. The Mantine hook really simplifies it a lot

But also, the `useEffect` part is actually there to update the color scheme if the user changes their system color preference after the app is already opened. With your other suggestion, I've currently simplified the code to this:

```ts
const prefersDarkMode = useMediaQuery("(prefers-color-scheme: dark)");
const [colorScheme, setColorScheme] = useState<"dark" | "light">(
    prefersDarkMode ? "dark" : "light"
);

useEffect(() => {
    setColorScheme(prefersDarkMode ? "dark" : "light");
}, [prefersDarkMode]);

const toggleColorScheme = () => {
    setColorScheme((prevScheme) => prevScheme === "light" ? "dark" : "light");
};
```

But regardless, we could still remove it if you think it's unnecessary or disruptive